libr/bin/p/bin_wad.c

```diff
@@ -0,0 +1,97 @@
+/* radare - LGPL3 - 2021 - murphy */
+
+#include <r_bin.h>
+#include <r_lib.h>
+#include "wad/wad.h"
+
+static WAD_Hdr loaded_header;
+
+static bool check_buffer(RBuffer *b) {
+	r_return_val_if_fail (b, false);
+	ut8 sig[4];
+	if (r_buf_read_at (b, 0, sig, sizeof (sig)) != 4) {
+		return false;
+	}
+	if (memcmp (sig, "IWAD", 4) && memcmp (sig, "PWAD", 4)) {
+		return false;
+	}
+    return true;
+}
+
+static bool load_buffer(RBinFile *bf, void **bin_obj, RBuffer *b, ut64 loadaddr, Sdb *sdb) {
+    if (r_buf_read_at (b, 0, (ut8*)&loaded_header, sizeof (loaded_header)) == sizeof (loaded_header)) {
+		*bin_obj = &loaded_header;
+		return true;
+	}
+    eprintf ("Truncated Header\n");
+	return false;
+}
+
+static RBinInfo *info(RBinFile *bf) {
+	RBinInfo *ret = R_NEW0 (RBinInfo);
+	if (!ret || !bf) {
+		return NULL;
```

Comment:
if bf == NULL and ret != NULL it's leaking here. better use r_return_val_if_fail (bf, NULL); before the R_NEW